server/src/main/java/org/opensearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -544,6 +554,11 @@ private IndexMetadata buildAndValidateTemporaryIndexMetadata(
         tmpImdBuilder.settings(indexSettings);
         tmpImdBuilder.system(isSystem);
 
+        if (remoteStoreBlobPathResolver != null) {
+            String pathType = remoteStoreBlobPathResolver.resolveType().toString();
+            tmpImdBuilder.putCustom(IndexMetadata.REMOTE_STORE_CUSTOM_KEY, Map.of(RemoteStoreBlobPathType.NAME, pathType));
```

Comment:
Do we need to handle backward compatibility with this change including node to node communication or snapshot restore?

Reply:
We are using existing fields in IndexMetadata for storing the remote store custom data for an index. I have validated node-to-node communication and snapshot (creation / restore) with the remote store custom data and it works fine. There are couple of follow up items where we need to associate the path type for following cases : 1. Index migration from document to segment replication type (using remote store) 2. Restoring a snapshot that is taken in older version or using document replication type.

Reply:
I am tracking the sub tasks (or issues) in the meta - https://github.com/opensearch-project/OpenSearch/issues/12589.